clang/test/CIR/CodeGen/multi-vtable.cpp

```diff
@@ -0,0 +1,107 @@
+// RUN: %clang_cc1 -triple x86_64-unknown-linux-gnu -std=c++20 -fclangir -mconstructor-aliases -clangir-disable-emit-cxx-default -emit-cir %s -o %t.cir
+// RUN: FileCheck --input-file=%t.cir %s
+
+class Mother {
+public:
+ virtual void MotherFoo() {}
+ void simple() { }
+ virtual void MotherFoo2() {}
+};
+
+class Father {
+public:
+ virtual void FatherFoo() {}
+};
+
+class Child : public Mother, public Father {
+public:
+ void MotherFoo() override {}
+};
+
+int main() {
+    Mother *b = new Mother();
+    b->MotherFoo();
+    b->simple();
+    Child *c = new Child();
+    c->MotherFoo();
+    c->MotherFoo2();
+    delete b;
+    delete c;
+
+    return 0;
+}
```

Comment:
Does this build all the way to LLVM? Would be nice to have LLVM tests right here as well (see other test for examples) and already include the LLVM lowering support needed (if any is necessary), so we can make sure that we are matching what's expected out from C++ source.

Reply:
Thanks! Updated.